### ISSUE TITLE
Vector2.slerp() returns normalized error 

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot#32692]
Godot version: 3.11

OS: Windows 7 Ultimate 64 bit

I keep getting these errors when trying to use slerp
```
E 0:00:03:0247   Condition ' !is_normalized() ' is true. returned: Vector2()
  <C Source>     ./core/math/vector2.h:234 @ slerp()
  <Stack Trace>  Player.gd:30 @ _process()```
```
To reproduce..
1. run this script..

```
extends Node2D

func _ready():
	pass # Replace with function body.

func _process(delta):
	#._process(delta);
	
	var vel 
	var max_vec = Vector2(4, 0).rotated(rotation);
	var min_vec = Vector2(0, 0).rotated(rotation);
	
	if   (Input.is_action_pressed("ui_down")):
		vel = Vector2(0,0).slerp(min_vec, 0.05);
	elif (Input.is_action_pressed("ui_up")):		
		vel = Vector2(0,0).slerp(max_vec, 0.01);

```
2. see output.

**Comment by Shadowblitz16** [comment]
I didn't realize most slerp methods took a value of 1.
if possible it should check if its less or greater then a normalized vector and if so do extra calculations do slerp between the two.

**Comment by hammeron-art** [comment]
Math types aim for performance. Checks and extra calculations hurt performance so the user code is responsible for passing valid arguments. The error is for debug purposes so you can find the problem and correct it but all validation checks are removed in the release build.

**Comment by Shadowblitz16** [comment]
couldn't the length be stored and added back like so?
```
func move(src, dst, speed):
    
    var _len = src.length();
    return (src.normalized()).slerp((dst.normalized()), speed) * _len;```

it doesn't seem like it would be that bad.
besides if performance is a issue optimizing GDScript itself would boost performance more then this would hurt it.

**Comment by Flavelius** [comment]
This is an issue with Vector3.Slerp too. Even though both vectors are passed in as .Normalized(), the IsNormalized() check inside Slerp fails; it should probably be removed anyway as the description clearly states that the inputs must be normalized, so it just unnecessarily wastes cpu cycles

**Comment by Calinou** [comment]
> it should probably be removed anyway as the description clearly states that the inputs must be normalized, so it just unnecessarily wastes cpu cycles.

If we removed the checks, people would be very confused as to why their vector calculations don't work as expected.

**Comment by Flavelius** [comment]
There seems to be an intended distinction between the naming of Slerp and linearInterpolate (lerp?) anyway, so the mentioned confusion would only reach as far as a look into the documentation to be sure of the distinction between those two. And using a method without knowing what it expects & produces always leads to confusion. But build-time skips are fine too (IsNormalized still needs to be adjusted/fixed though).
shouldn't it be enough to check Mathf.Abs(lengthsquared-1f) < float.epsilon;  without the extra cast to double?

**Comment by akien-mga** [comment]
Fixed by #55877.
